nixos/modules/security/doas.nix

```diff
@@ -0,0 +1,266 @@
+                restricted PATH will be searched. When set to <code>null</code>, all commands are
+                allowed.
+              '';
+            };
+
+            args = mkOption {
+              type = with types; nullOr (listOf str);
+              default = [];
+              description = ''
+                Arguments to the command. The arguments provided by the user need to match those
+                specified. When set to <code>null</code>, the command must be run without any
+                arguments.
+              '';
+            };
+          };
+        }
+      );
+    };
+
+    extraConfig = mkOption {
+      type = with types; lines;
+      default = "";
+      description = ''
+        Extra configuration text appended to <filename>doas.conf</filename>.
+      '';
+    };
+  };
+
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+
+    security.doas.extraRules = [
+      {
+        groups = [ "wheel" ];
+        noPass = !cfg.wheelNeedsPassword;
+      }
+    ];
+
+    # TODO setenv SSH_AUTH_SOCK?
+    security.doas.configFile = ''
+      # Don't edit this file. Set the NixOS options ‘security.doas.configFile’
+      # or ‘security.doas.extraRules’ instead.
+
+      # "root" is allowed to do anything.
+      permit nopass keepenv root
+
+      # extraRules
+      ${concatStringsSep "\n" (
+      lists.flatten (
+        map (
+          rule:
+            let
+              opts = concatStringsSep " " [
+                (optionalString rule.noPass "nopass")
+                (optionalString rule.persist "persist")
+                (optionalString rule.keepEnv "keepenv")
+                (optionalString (length rule.setEnv > 0) "setenv { ${concatStringsSep " " rule.setEnv} }")
+              ];
+
+              as = optionalString (!isNull rule.runAs) "as ${rule.runAs}";
+
+              cmd = optionalString (!isNull rule.cmd) "cmd ${rule.cmd}";
+
+              args = if (isNull rule.args) then
+                "args"
+              else if (length rule.args > 0) then
+                "args ${concatStringsSep " " rule.args}"
+              else "";
+            in
+              optionals (length cfg.extraRules > 0) [
+                (
+                  optionalString (length rule.users > 0)
+                    (map (usr: "permit ${opts} ${mkUsrString usr} ${as} ${cmd} ${args}") rule.users)
+                )
+                (
+                  optionalString (length rule.groups > 0)
+                    (map (grp: "permit ${opts} ${mkGrpString grp} ${as} ${cmd} ${args}") rule.groups)
+                )
+              ]
+        ) cfg.extraRules
+      )
+    )}
+
+      # extraConfig
+      ${cfg.extraConfig}
+    '';
+
+    security.wrappers = {
+      doas.source = "${doas}/bin/doas";
+    };
+
+    environment.systemPackages = [
+      doas
+    ];
+
+    security.pam.services.doas = {
+      allowNullPassword = true;
+      sshAgentAuth = true;
```

Comment:
`sshAgentAuth` should not be enabled by default?

Reply:
I pushed a commit removing this, but then went back and noticed that the `sudo` module [does indeed set `sshAgentAuth` to true](https://github.com/NixOS/nixpkgs/blob/cb1f1b42602a2655df69d7690f428b4d8467901a/nixos/modules/security/sudo.nix#L215)... Thoughts?

Reply:
Ok! Let's default to the same value as `sudo`, it's probably the least surprising option.